.github/actions/build_pandas/action.yml

```diff
@@ -3,15 +3,28 @@ description: Rebuilds the C extensions and installs pandas
 runs:
   using: composite
   steps:
-
     - name: Environment Detail
       run: |
-        conda info
-        conda list
+        if which mamba 2>/dev/null; then
+          mamba info
+          mamba list
+          mamba info | grep -Ei 'environment.+:' | grep -qEiv 'environment.+:.+none'
+        fi
+        python -VV
+        which python
+        which pip
+        pip list
       shell: bash -el {0}
 
     - name: Build Pandas
       run: |
-        python setup.py build_ext -j 2
-        python -m pip install -e . --no-build-isolation --no-use-pep517 --no-index
+        time python setup.py build_ext -v -j $N_JOBS
+        pip install -v -e . --no-build-isolation --no-use-pep517 --no-index
+      shell: bash -el {0}
+      env:
+        # Cannot use parallel compilation on Windows, see https://github.com/pandas-dev/pandas/issues/30873
+        N_JOBS: ${{ runner.os == 'Windows' && 1 || 3 }}
```

Comment:
Why 3 jobs if not windows? My understanding is that GItHub runners have 2 cores.

Reply:
macOS has 3, can change that back to 2 and change to 3 in a follow-up PR where we add macOS?

Reply:
Or we could just hardcode Windows 1, Linux 2, macOS 3 here?